src/galario.cpp

```diff
@@ -1042,11 +1042,29 @@ void create_image_d(int nr, const dreal* const ints, dreal Rmin, dreal dR, int n
 
     auto const nblocks = (2*rmax) / tpb + 1;
     sweep_d<<<dim3(nblocks, nblocks), dim3(tpb, tpb)>>>(nr, ints_d, Rmin, dR, rmax, nxy, dxy, inc, sr_to_px, *addr_image_d);
-    CCheck(cudaDeviceSynchronize());
+    CCheck(cudaDeviceSynchronize()); // TODO: can be moved after central pixel computation?
```

Comment:
I think you may want to repeat the `CCheck(cudaDeviceSynchronize());` after `central_pixel_d`. I think the reason for introducing it was that an error in a kernel call doesn't signal anything so we check explicitly. If you moved the check to after `central_pixel_d`, we could not know if an error occured in that kernel or the previous kernel `sweep_d`

Reply:
Good point